lib/ansible/plugins/callback/logdna.py

```diff
@@ -77,128 +226,342 @@ def get_ip():
+        call logdna helper method
+        """
+        action = result._task_fields.get('action')
+        role = result._task._role
+        play = self.logdna_callback.ansible_playbook
+        if not ((action in self.conf_ignore_actions)
+                or (role in self.conf_ignore_roles)
+                or (play in self.conf_ignore_plays)
+                or (status.lower() in self.conf_ignore_statuses)
+                or (self.disabled)):
+            self.logdna_callback.send_logdna(
+                self.conf_appname,
+                self.conf_endpoint,
+                self.conf_disable_loglevel,
+                self.conf_host,
+                self.conf_hostname,
+                self.conf_ingestion_key,
+                self.conf_ip_addr,
+                self.conf_log_fmt,
+                self.conf_mac_addr,
+                self.conf_tags,
+                self.conf_timeout,
+                status,
+                result,
+                self._execution_timer(result)
+            )
 
     def set_options(self, task_keys=None, var_options=None, direct=None):
-        super(CallbackModule, self).set_options(task_keys=task_keys, var_options=var_options, direct=direct)
-
-        self.conf_key = self.get_option('conf_key')
-        self.plugin_ignore_errors = self.get_option('plugin_ignore_errors')
-        self.conf_hostname = self.get_option('conf_hostname')
-        self.conf_tags = self.get_option('conf_tags')
-        self.mac = get_mac()
-        self.ip = get_ip()
+        """
+        setup the callback options
+        """
+        super(CallbackModule, self).set_options(
+            task_keys=task_keys, var_options=var_options, direct=direct)
+
+        self.conf_appname = self.get_option('logdna_appname')
+        if self.conf_appname is None:
+            self.conf_appname = self.defaults.get('logdna_appname')
+
+        self.conf_endpoint = self.get_option('logdna_endpoint')
+        if self.conf_endpoint is None:
+            self.conf_endpoint = self.defaults.get('logdna_endpoint')
+
+        self.conf_disable_loglevel = self.get_option(
+            'logdna_disable_loglevels')
+
+        self.conf_host = self.get_option('logdna_host')
+        if self.conf_host is None:
+            self.conf_host = self.defaults.get('logdna_host')
+
+        self.conf_timeout = self.get_option('logdna_timeout')
+        if not self.conf_timeout:
+            self.conf_timeout = self.defaults.get('logdna_timeout')
+        # if use_local_hostname is set to any value, use get_local_hostname()
+        # of the ansible runner host, else default to logdna_hostname and
+        # if that is unset, the targeted ansible host will be used.
+        self.conf_use_target_host_for_hostnmae = self.get_option(
+            'logdna_use_target_host_for_hostname')
+        self.conf_hostname = self.get_option('logdna_hostname')
+        if not self.conf_use_target_host_for_hostnmae:
+            self.conf_hostname = get_local_hostname()
```

Comment:
worth noting that if there is no hostname, or if mac/ip is in an invalid format, the request will fail